BoxBrowseSDK/BoxBrowseSDK/BOXFolderViewController.m

```diff
@@ -90,16 +94,35 @@ - (void)viewDidLoad
 
     [self setupForSearch];
     
-    UILabel *emptyLabel = [[UILabel alloc] init];
-    emptyLabel.textAlignment = NSTextAlignmentCenter;
-    emptyLabel.font = [UIFont boldSystemFontOfSize:15.0f];
-    emptyLabel.textColor = [UIColor colorWithRed:180.0f/255.0f green:179.0f/255.0f blue:180.0f/255.0f alpha:1.0f];
-    emptyLabel.backgroundColor = [UIColor clearColor];
-    emptyLabel.contentMode = UIViewContentModeCenter;
-    emptyLabel.numberOfLines = 0;
-    emptyLabel.autoresizingMask = UIViewAutoresizingFlexibleWidth;
-    emptyLabel.text = NSLocalizedString(@"Loading…", @"Loading message shown while content is being fetched.");
-    self.tableView.backgroundView = emptyLabel;
+    self.activityIndicator = [[UIActivityIndicatorView alloc] initWithActivityIndicatorStyle:UIActivityIndicatorViewStyleGray];
+    self.activityIndicator.hidesWhenStopped = YES;
+    [self.activityIndicator startAnimating];
+
+    self.folderInfoLabel = [[UILabel alloc] init];
+    self.folderInfoLabel.textAlignment = NSTextAlignmentCenter;
+    self.folderInfoLabel.font = [UIFont boldSystemFontOfSize:15.0f];
+    self.folderInfoLabel.textColor = [UIColor colorWithRed:180.0f/255.0f green:179.0f/255.0f blue:180.0f/255.0f alpha:1.0f];
+    self.folderInfoLabel.backgroundColor = [UIColor clearColor];
+    self.folderInfoLabel.contentMode = UIViewContentModeCenter;
+    self.folderInfoLabel.numberOfLines = 0;
+    self.folderInfoLabel.text = @"";
+    self.folderInfoLabel.hidden = YES;
+    self.folderInfoLabel.autoresizingMask = UIViewAutoresizingFlexibleWidth;
+}
+
+- (void)viewDidLayoutSubviews
+{
+    [super viewDidLayoutSubviews];
+
+    self.backgroundView = [[UIView alloc] initWithFrame:self.view.bounds];
```

Comment:
Minor: we could move subview-specific layout code into separate methods so they can be triggered elsewhere, if needed.